### PR TITLE
Support search and info for genes without phenotype annotations in switch to Monarch data.

### LINKED ISSUE #398: gene search and gene info don't work correctly for genes without phenotypes
In the new pipeline, genes without phenotypes aren't linked to the type `http://purl.org/phenoscape/vocab.owl#AnnotatedGene`. This is used in the queries for the above services, causing no results to be returned. The solution to this may be a change in either the queries, or in the pipeline.

### PR DESCRIPTION
**Pull request by balhoff**
Fixes #398

### CODE REVIEW

File: src/main/scala/org/phenoscape/kb/Main.scala [hunk 1]
```diff
@@ -1009,10 +1008,10 @@ object Main extends HttpApp with App {
               } ~
               pathPrefix("gene") {
                 path("search") {
-                  parameters('text, 'taxon.as[IRI].?) { (text, taxonOpt) => //FIXME add limit option?
+                  parameters('text, 'taxon.as[IRI].?, 'limit.as[Int].?(100)) { (text, taxonOpt, limit) =>
```
Comment (Shalsh23) [inline src/main/scala/org/phenoscape/kb/Main.scala:1011]: Swagger yaml file would need to be updated with this parameter.

File: src/main/scala/org/phenoscape/kb/Gene.scala [hunk 2]
```diff
@@ -61,20 +61,33 @@ object Gene {
...
-    query.setLimit(100)
-    query
+    val queryLimit = limit.map(l => sparql"LIMIT $l").getOrElse(sparql"")
+    val query =
+      sparql"""
+            SELECT DISTINCT ?gene ?gene_label ?taxon ?taxon_label
+            FROM $KBMainGraph
+            WHERE {
+              ?gene_label $BDSearch $searchText .
+              ?gene_label $BDMatchAllTerms true .
```
Comment (Shalsh23) [inline src/main/scala/org/phenoscape/kb/Gene.scala:73]: Does the literal `true` need to be `Boolean` or `String`?

Comment (balhoff) [reply src/main/scala/org/phenoscape/kb/Gene.scala:73]: It is automatically a boolean if written this way.

File: src/main/scala/org/phenoscape/kb/Main.scala (hunk 1 above)
Comment (balhoff) [reply src/main/scala/org/phenoscape/kb/Main.scala:1011]: Fixed.

**Review by Shalsh23 — APPROVED** [review]

**Pull request merged**